minecraft-plugin/src/main/java/com/pvchat/proximityvoicechat/plugin/distanceMatrix/PlayerDistanceAndVolumeCalculations.java

```diff
@@ -0,0 +1,122 @@
+package com.pvchat.proximityvoicechat.plugin.distanceMatrix;
+
+import com.pvchat.proximityvoicechat.plugin.ProximityVoiceChat;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class PlayerDistanceAndVolumeCalculations {
+
+    public List<Player> players;
+    private ProximityVoiceChat pluginInstance;
+    private HashMap<String, List<Player>> playersToWorld;
+    private ArrayList<Consumer<List<PlayerVolumeData>>> stateChangeListeners;
+
+    public PlayerDistanceAndVolumeCalculations(ProximityVoiceChat pluginInstance) {
+        this.pluginInstance = pluginInstance;
+        players = new ArrayList<>();
+        playersToWorld = new HashMap<>();
+        stateChangeListeners = new ArrayList<>();
+    }
+
+    public void addStateChangeListener(Consumer<List<PlayerVolumeData>> stateChangeListener){
+        stateChangeListeners.add(stateChangeListener);
+    }
+
+    public void removeStateChangeListener(Consumer<List<PlayerVolumeData>> stateChangeListener) {
+        stateChangeListeners.remove(stateChangeListener);
+    }
+
+    //Getting Online players
+    public void getPlayers() {
+        players.clear();
+        players.addAll(Bukkit.getOnlinePlayers().stream().toList());
+
+        playersToWorld.clear();
+
+        for(World w : Bukkit.getServer().getWorlds()){
+            playersToWorld.put(w.getName(), new ArrayList<>());
+        }
+
+        //Dividing per realm
+        for (Player temp : players) {
+            String world = temp.getWorld().getName();
+
+            playersToWorld.get(world).add(temp);
+        }
+    }
+
+    private void addToVolumeList(List<PlayerVolumeData>  playerVolumeList, List<Player>  playersOfSomeWorld){
+        int size = playersOfSomeWorld.size();
+        if (size > 1) {
+            for (int i = 0; i < size; i++) {
+                for (int j = i + 1; j < size; j++) {
+                    Player p1, p2;
+                    p1 = playersOfSomeWorld.get(i);
+                    p2 = playersOfSomeWorld.get(j);
+                    double distance = distanceCalculator(p1.getLocation(), p2.getLocation());
+                    long volume = calculateVolume(distance);
+                    if (volume != -1) {
+                        PlayerVolumeData temporary = new PlayerVolumeData(p1.getUniqueId().toString(),p2.getUniqueId().toString(),volume);
+                        playerVolumeList.add(temporary);
+                    }
```

Comment:
Introducing such "special values" is misleading. If you want a function to return results only for some values of the arguments then you should model it with an `Optional`.